### PR TITLE
User controller spec (part 1)

### PR DESCRIPTION
**Pull request by mdeniz**
Just started with user controller, it was getting too complex and just wanted to split the work

### CODE REVIEW

File: src/api/spec/views/webui/user/show.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
```
Comment (hennevogel) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: Why do you add a view test for this?

Comment (mdeniz) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: Just transcribed to the proper type of spec... we can get rid of it if you prefer that

Comment (hennevogel) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: Yeah I wouldn't start testing views. It's not worth it...

Comment (mdeniz) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: Should I transform it to a feature spec? Is that better?

Comment (hennevogel) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: No I don't think so. I mean what are you testing with this test? That `unless` doesn't break?

Comment (mdeniz) [inline src/api/spec/views/webui/user/show.html.erb_spec.rb]: NO!!! hahaha. Just testing that 'when the current user is nobody' it 'hides the email' and  'when the current user is valid'  it "displays the email",..., no matter the view code... just that works like that

**Comment by mdeniz** [comment]
@hennevogel I removed the view spec.
@ChrisBr and @bgeuken please review it.

File: src/api/spec/rails_helper.rb [hunk 2]
```diff
@@ -26,6 +26,10 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+  config.expect_with :rspec do |c|
+    c.syntax = :expect  # just use the new syntax
```
Comment (mdeniz) [inline src/api/spec/rails_helper.rb]: @ChrisBr this is the same as in your PR, but you did it in the spec_helper.rb file... So if yours get merged before mine I will drop this change

File: src/api/spec/controllers/webui/user_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,184 @@
+require 'rails_helper'
+
+RSpec.describe Webui::UserController do
+  let!(:user) { create(:confirmed_user, login: "tom") }
+  let!(:non_admin_user) { create(:confirmed_user, login: "moi") }
+  let!(:admin_user) { create(:admin_user, login: "king") }
+  let(:deleted_user) { create(:deleted_user) }
+
+  describe "GET #index" do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: This is already tested in [spec/controllers/webui/webui_controller_spec.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/webui_controller_spec.rb) you should only test that the before filter is used like in [spec/controllers/webui/configuration_controller_spec.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/configuration_controller_spec.rb#L11)

Comment (mdeniz) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: Ok, I will change it to remove the duplication.

Comment (mdeniz) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: @hennevogel using   `use_before_action(:require_admin) }` only ensures that we have this before_action/filter defined in the controller... nothing to do with checking if the filter was running inside an action.
If the `before_action` have `:except` or `:only` options... then you can't check with this if the filter was running or not in your example after a http call

Comment (mdeniz) [inline src/api/spec/controllers/webui/user_controller_spec.rb]: @hennevogel @ChrisBr: So doing this https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/controllers/webui/configuration_controller_spec.rb#L8 is a non sense, the checking of the before_action should be outside any "describe" and then inside each action you should check the templated rendered at least

**Comment by hennevogel** [comment]
Aweomse :cake:

**Comment by bgeuken** [comment]
@mdeniz @hennevogel Why was this commit merged? ->  3c6fab4bfff71b69d1af060a7c0a05cded03ef6a

Next time please rebase and remove such things.

**Pull request merged**